src/lammpsio/data.py

```diff
@@ -280,7 +390,7 @@ def read(self):
             elif None in box_bounds:
                 raise IOError("Box bounds not read")
             box = Box(box_bounds[:3], box_bounds[3:], box_tilt)
-            snap = Snapshot(N, box)
+            snap = Snapshot(N, box, num_types=num_types)
```

Comment:
`Snapshot`'s constructor is `def __init__(self, N, box, step=None, num_types=None):`, so I think this one does have to have a kwarg argument since we're not passing a step argument. 

This change works fine for the topology types.

Reply:
You are right! I forgot about the `step` argument. A kwarg is required here.